CMakeLists.txt

```diff
@@ -1,88 +1,45 @@
-cmake_minimum_required(VERSION 3.16)
-project(jluna)
+cmake_minimum_required(VERSION 3.12)
```

Comment:
I could add the 3.22 dependency, jluna is already using the newest compilers, might as well make all of it as modern as possible

Reply:
I don't see a reason to unnecessarily raise the requirement. `PROJECT_IS_TOP_LEVEL` is trivial to backport.

Reply:
I agree, I figured that would be the best way of doing things